cloudflare/resource_cloudflare_custom_ssl.go

```diff
@@ -0,0 +1,284 @@
+package cloudflare
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareCustomSsl() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareCustomSslCreate,
+		Read:   resourceCloudflareCustomSslRead,
+		Update: resourceCloudflareCustomSslUpdate,
+		Delete: resourceCloudflareCustomSslDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareCustomSslImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"zone_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"custom_ssl_priority": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"priority": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"custom_ssl_options": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"certificate": {
+							Type:     schema.TypeString,
+							Optional: false,
+						},
+						"private_key": {
```

Comment:
We might also want to use [`Sensitive`](https://github.com/hashicorp/terraform/blob/master/helper/schema/schema.go#L249) on this field property to prevent it from being displayed in output/logs.

Reply:
Agreed